platform/security/idp/security-idp-server/src/test/java/org/codice/ddf/security/idp/server/IdpEndpointTest.java

```diff
@@ -0,0 +1,563 @@
+        String relayState = "94697cdc-e64f-4edf-b26a-52c14c2314dd";
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        Response response = idpEndpoint.showPostLogin(samlRequest, relayState, request);
+        assertThat(response.getEntity().toString(), containsString("SAMLRequest"));
+        assertThat(response.getEntity().toString(), containsString("RelayState"));
+        assertThat(response.getEntity().toString(), containsString("ACSURL"));
+    }
+
+    @Test
+    public void testShowGetLogin() throws WSSecurityException {
+        String samlRequest = "jZLNbsIwEIRfJfLdJHEIDRZBoqCqSLSiQHvopTLOApYcO/U6/Xn7JoFK9EK52uNvZsc7QlFqVvFJ7Q9mBe81oA8miOC8smZqDdYluDW4DyXhebXIycH7CnkYaiuFPlj0PBsOkxCPEgxbYIhoSTBrWMqIFvT/M1VUzd1eGRLMZzl5E2k62CZyR7cDkLQfJYxu07hPAbJdVkhWsDhppIg1zA16YXxOWBSnNI4oG27iG84SzrJeFEWvJFg66620+laZQpl9TmpnuBWokBtRAnIv+XrysOCsF/HtUYT8frNZ0hUUyoH0JHgBh90sjYgEX6U2yI/1XeZVJ3MyHnVy3qV254TLAPH7H2R8Tfuj8NzmZFrxx4Y7ny2tVvI7mGhtP6cOhIeceFcDCe6sK4W/nKQ9UQXddVJetYWgB9OUs162/KdaaLVT4K7bExKOT2H/buD4Bw==";
+        String relayState = "ef95c04b-6c05-4d12-b65f-dd32fed8811e";
+        String signatureAlgorithm = "http://www.w3.org/2000/09/xmldsig#rsa-sha1";
+        String signature = "CQyv7hn001ClryULPk+DCqMiZYFnpmnVneP2byUoQ8rQr2fkM9F7y5f7CsIyyO8+uLO0ffXLHy7OrmnnbAeKJ2hRUJLUl8dt5NWqDPjGLfNCA8KshSeoxhbwc5PS3zjIGzdox/rdbsDJ4BOL2Jyh2OzlVKry+QAh3wmJDQrFcxs=";
+
+        HttpServletRequest request = mock(HttpServletRequest.class);
+        Response response = idpEndpoint
+                .showGetLogin(samlRequest, relayState, signatureAlgorithm, signature, request);
+        assertThat(response.getEntity().toString(), containsString("SAMLRequest"));
+        assertThat(response.getEntity().toString(), containsString("RelayState"));
+        assertThat(response.getEntity().toString(), containsString("ACSURL"));
+    }
+
+    @Test
+    public void testRetrieveMetadata() throws WSSecurityException, CertificateEncodingException {
+        Response response = idpEndpoint.retrieveMetadata();
+        assertThat(response.getEntity().toString(), containsString("IDPSSODescriptor"));
+        assertThat(response.getEntity().toString(), containsString("SingleLogoutService"));
+        assertThat(response.getEntity().toString(), containsString("SingleSignOnService"));
+    }
+
+    @Test
+    public void testProcessLoginBasic() {
+        String samlRequest = "jZLNbsIwEIRfJfLdJHEIDRZBoqCqSLSiQHvopTLOApYcO/U6/Xn7JoFK9EK52uNvZsc7QlFqVvFJ7Q9mBe81oA8miOC8smZqDdYluDW4DyXhebXIycH7CnkYaiuFPlj0PBsOkxCPEgxbYIhoSTBrWMqIFvT/M1VUzd1eGRLMZzl5E2k62CZyR7cDkLQfJYxu07hPAbJdVkhWsDhppIg1zA16YXxOWBSnNI4oG27iG84SzrJeFEWvJFg66620+laZQpl9TmpnuBWokBtRAnIv+XrysOCsF/HtUYT8frNZ0hUUyoH0JHgBh90sjYgEX6U2yI/1XeZVJ3MyHnVy3qV254TLAPH7H2R8Tfuj8NzmZFrxx4Y7ny2tVvI7mGhtP6cOhIeceFcDCe6sK4W/nKQ9UQXddVJetYWgB9OUs162/KdaaLVT4K7bExKOT2H/buD4Bw==";
```

Comment:
It would be nice to have these requests loaded from their own XML file, as it's duplicated multiple times.